src/awkward/_nplikes/__init__.py

```diff
@@ -27,6 +28,21 @@ def to_nplike(
     if from_nplike is nplike:
         return array
 
+    if isinstance(array, awkward._nplikes.virtual.VirtualArray):
+        if not array.is_materialized:
+            raise TypeError(
+                "Cannot convert a VirtualArray to a different nplike without materializing it first. Use ak.materialize on the array to do so."
+            )
+        else:
+            if isinstance(
+                nplike, (awkward._nplikes.numpy.Numpy, awkward._nplikes.cupy.Cupy)
+            ):
+                array = array.materialize()
+            else:
+                raise TypeError(
+                    f"Only numpy and cupy nplikes are supported for VirtualArray. Received {type(nplike)}"
+                )
+
```

Comment:
Can't we just set the `nplike` of the virtual array to a different one without materialization? The `nplike` property on the virtual array should make sure that it is only supported for numpy and cupy.